meteor/sofie-server/data-access/repositories/mongo/mongo-segment-repository.ts

```diff
@@ -33,4 +35,22 @@ export class MongoSegmentRepository extends BaseMongoRepository implements Segme
 			})
 		)
 	}
+
+	public async deleteSegmentsForRundown(rundownId: string): Promise<void> {
+		this.assertDatabaseConnection(this.deleteSegmentsForRundown.name)
+		const segments: Segment[] = await this.getSegments(rundownId)
+
+		segments.forEach(async (segment: Segment) => this.partRepository.deletePartsForSegment(segment.id))
```

Comment:
You can avoid the "Promise returned from forEach argument is ignored" by using:
`
for (const segment of segments) {
	await this.partRepository.deletePartsForSegment(segment.id)
}
`

Reply:
Has been changed in commit https://github.com/tv2/tv-automation-server-core/pull/255/commits/34447718d76f7e135963c55c38a9b220a97f3416.